components/generator/input.vue

```diff
@@ -42,21 +51,46 @@ export default {
   },
   computed: {
     ...mapState('generator', [
-      'shortcut'
-    ])
+      'shortcut',
+      'isGroup'
+    ]),
+    ...mapGetters('generator', [
+      'qualifiedShortcut'
+    ]),
+    submitIcon () {
+      return this.isGroup
+        ? 'plus-circle'
+        : 'chevron-circle-right'
+    }
   },
   methods: {
     ...mapActions('generator', [
-      'generate'
+      'generate',
+      'addLink',
+      'setIsGroup'
+    ]),
+    ...mapActions('snackbar', [
+      'showSnackbar'
     ]),
+
     create () {
       this.validateInput()
 
       if (!this.showWarning) {
-        this.generate([this.target]).then(() => {
-          storageService.addToHistory(this.shortcut)
-          this.$router.push({ path: '/share' })
-        })
+        this.addLink(this.target)
+        this.target = ''
+
+        if (!this.isGroup) {
+          this.generate().then(() => {
```

Comment:
It would be good to find a way to not have this code duplicated between `group` and `input`